### PR TITLE
Hotfix/bug fixes

### PR DESCRIPTION
**Pull request by cpviolator**
The PR fixes some bugs in the test utils spinor construction code, as well as cleanly deleting the `cloverEigensolver` field.

### CODE REVIEW

**Comment by weinbe2** [comment]
Reviewing now. Can you please hit this with a `clang-format` @cpviolator ?

File: utils/host_utils.cpp [hunk 1]
```diff
@@ -221,10 +228,10 @@ void constructWilsonTestSpinorParam(quda::ColorSpinorParam *cs_param, const Quda
     cs_param->nDim = 4;
   }
   for (int d = 0; d < 4; d++) cs_param->x[d] = gauge_param->X[d];
-  bool pc = (inv_param->solution_type == QUDA_MATPC_SOLUTION || inv_param->solution_type == QUDA_MATPCDAG_MATPC_SOLUTION);
+  bool pc = isPCSolution(inv_param->solution_type);
```
Comment (weinbe2) [inline utils/host_utils.cpp:224]: This also needs to be propagated into `utils/staggered_host_utils.cpp`

File: utils/host_utils.cpp [hunk 2]
```diff
@@ -146,6 +146,13 @@ void setQudaDefaultMgTestParams()
   }
 }
 
+bool isPCSolution(QudaSolutionType solution_type) {
```
Comment (weinbe2) [inline utils/host_utils.cpp]: This definition could be moved into `utils/host_utils.h` and then get used in `utils/staggered_host_utils.cpp`

**Comment by weinbe2** [comment]
Minus my couple of comments on propagating this to `utils/staggered_host_utils.cpp` this looks good. Great catches all around.

We should propagate this into `lib/interface_quda.cpp` in the future, but for now let's get these bugfixes in.

**Review by weinbe2 — APPROVED** [review]
Approved pending staggered requests, clang-format, and Jenkins.

File: utils/host_utils.cpp (hunk 2 above)
Comment (cpviolator) [reply utils/host_utils.cpp]: 7052e04

File: utils/host_utils.cpp (hunk 1 above)
Comment (cpviolator) [reply utils/host_utils.cpp:224]: 7052e04

**Comment by cpviolator** [comment]
@weinbe2 @maddyscientist Thank you both!

**Comment by weinbe2** [comment]
Thanks for the quick fix @cpviolator , looks good!

**Pull request merged**